reference-controllers/zookeeper-controller/src/main.rs

```diff
@@ -217,48 +217,51 @@ async fn reconcile_stateful_set(zk: &ZookeeperCluster, client: Client) -> Result
             ..old_sts
         };
         // Scale up or no change
-        sts_api
-            .replace(sts_name, &PostParams::default(), &updated_sts)
-            .await
-            .map_err(Error::ReconcileStatefulSetFailed)?;
-
         if updated_sts.spec.as_ref().unwrap().replicas.unwrap()
             < old_sts.spec.as_ref().unwrap().replicas.unwrap()
         {
             // Scale down
             info!("Scale down statefulset: {}", sts_name);
 
-            let cluster = zk.metadata.name.clone().unwrap()
-                + "-client."
-                + zk.metadata.namespace.as_ref().unwrap()
-                + ".svc.cluster.local:2181";
-            info!("Connecting to Zookeeper cluster: {}", cluster);
-            let client = zk::Client::connect(cluster.as_str(), Duration::from_secs(20))
-                .await
-                .map_err(Error::ZkClusterConnectFailed)?;
-
-            let (config_bytes, stat, watcher) = client.get_and_watch_config().await.unwrap();
-
-            let input = String::from_utf8(config_bytes).unwrap();
-            let num_servers_to_keep = updated_sts.spec.as_ref().unwrap().replicas.unwrap() as usize;
-
-            // split
-            let mut servers: Vec<&str> = input.split('\n').collect();
-            servers.pop(); // the last one is version, discard
-
-            let kept_servers: Vec<&str> = servers.into_iter().take(num_servers_to_keep).collect();
-
-            // use ',' to join
-            let final_config = kept_servers.join(",");
-
-            info!("Modified config: {}", final_config);
-            let strings = vec![final_config.as_str()];
-            let test_update = EnsembleUpdate::New {
-                ensemble: strings.iter().cloned(),
+            let path = cluster_size_zk_node_path(zk);
+            info!("Try to create {} node since all replicas are ready", path);
```

Comment:
Are we waiting for all replica pods to be ready?

Reply:
Copy paste error